cli/waiter/token_post.py

```diff
@@ -272,12 +272,15 @@ def add_implicit_arguments(unknown_args, parser):
     for i in range(num_unknown_args):
         arg = unknown_args[i]
         if arg.startswith(("-", "--")):
+            arg_dest = arg.lstrip('-')
             if any(arg.endswith(suffix) for suffix in INT_PARAM_SUFFIXES):
                 arg_type = possible_int
             elif any(arg.endswith(suffix) for suffix in FLOAT_PARAM_SUFFIXES):
                 arg_type = possible_float
+            elif any(arg_dest.startswith(prefix) for prefix in STRING_PARAM_PREFIXES):
```

Comment:
Should we generalize this to nested field access instead?

Reply:
I wanted to be specific here to `env` and `metadata` because we are sure those values should be parsed as strings. There may be other nested fields that we add that have boolean/number values, and I didn't want to change the default behavior for other fields